lib/improver/tests/spotdata/spotdata/test_framework.py

```diff
@@ -0,0 +1,278 @@
+        iris.save(temperature_on_height_levels, ad_path_temperature)
+        iris.save(pressure_on_height_levels, ad_path_pressure)
+        iris.save(surface_pressure, ad_path_s_pressure)
+
+        self.config_path = self.configs + '/spotdata_constants.json'
+        ff = open(self.config_path, 'w')
+        json.dump(diagnostic_recipe, ff, sort_keys=True, indent=4,
+                  separators=(',', ': ',))
+        ff.close()
+
+        self.made_files = [data_path, orography_path, land_path,
+                           ad_path_temperature, ad_path_pressure,
+                           ad_path_s_pressure, self.config_path]
+
+        self.cube = cube
+        self.ancillary_data = ancillary_data
+        self.additional_data = additional_data
+        self.sites = sites
+        self.neighbour_list = neighbour_list
+        self.diagnostic_recipe = diagnostic_recipe
+
+        self.args = (self.config_path,
+                     self.data_directory,
+                     self.ancillary_directory)
+
+        site_properties = {
+            'latitudes': [10, 20, 30, 40, 50, 60],
+            'longitudes': [0, 0, 0, 0, 0, 0],
+            'altitudes': [0, 1, 2, 3, 4, 5],
+            'site_ids': [0, 1, 2, 3, 4, 5],
+            }
+
+        self.kwargs = {
+            'diagnostic_list': ['temperature'],
+            'site_path': None,
+            'constants_file_path': None,
+            'site_properties': site_properties,
+            'forecast_date': '20170217',
+            'forecast_time': 6,
+            'forecast_length': 2,
+            'output_path': self.output_directory,
+            'use_multiprocessing': False
+            }
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        for a_file in self.made_files:
+            Call(['rm', a_file])
+        Call(['rmdir', self.ancillary_directory])
+        Call(['rmdir', self.data_directory])
+        Call(['rmdir', self.configs])
+        Call(['rmdir', self.output_directory])
+
+
+class Test_missing_data(Test_run_framework):
+    """
+    Test the run_framework interface with various options.
+
+    """
+    def test_no_data_files(self):
+        """
+        Test framework raises an error when data files are not found at the
+        given data path.
+
+        """
+        msg = 'No relevant data files found'
+        with self.assertRaisesRegexp(IOError, msg):
+            Plugin(self.config_path, '', self.ancillary_directory,
+                   **self.kwargs)
+
+    def test_no_site_data(self):
+        """
+        Test framework raises an error when no SpotData site information is
+        provided.
+
+        """
+        self.kwargs['site_properties']['latitudes'] = None
+        self.kwargs['site_properties']['longitudes'] = None
+        msg = 'No SpotData site information has been provided'
+        with self.assertRaisesRegexp(ValueError, msg):
+            Plugin(*self.args, **self.kwargs)
+
+
```

Comment:
Could you add a test for the `Exception` at the end of `process_diagnostic` (see comment in `framework.py`).